doc/source/user/basics.creation.rst

```diff
@@ -66,43 +111,212 @@ examples will be given here: ::
+and the rest are 0, as such::
+
+ >>> np.eye(3)
+ array([[1., 0., 0.],
+        [0., 1., 0.],
+        [0., 0., 1.]])
+ >>> np.eye(3, 5)
+ array([[1., 0., 0., 0., 0.],
+        [0., 1., 0., 0., 0.],
+        [0., 0., 1., 0., 0.]])
+
+:func:`numpy.diag` can define either a square 2D array with given values along
+the diagonal _or_ if given a 2D array returns a 1D array that is
+only the diagonal elements. The two array creation functions can be helpful while
+doing linear algebra, as such::
+ 
+ >>> np.diag([1, 2, 3])
+ array([[1, 0, 0],
+       [0, 2, 0],
+       [0, 0, 3]])
+ >>> np.diag([1, 2, 3], 1)
+ array([[0, 1, 0, 0],
+        [0, 0, 2, 0],
+        [0, 0, 0, 3],
+        [0, 0, 0, 0]])
+ >>> a = np.array([[1, 2], [3, 4]])
+ >>> np.diag(a)
+ array([1, 4])
+
+``vander(x, n)`` defines a Vandermonde matrix as a 2D NumPy array. Each column
+of the Vandermonde matrix is a decreasing power of the input 1D array or
+list or tuple,
+``x`` where the highest polynomial order is ``n-1``. This array creation
+routine is helpful in generating linear least squares models, as such::
+ 
+ >>> np.vander(np.linspace(0, 2, 5), 2)
+ array([[0.  , 0.  , 1.  ],
+        [0.25, 0.5 , 1.  ],
+        [1.  , 1.  , 1.  ],
+        [2.25, 1.5 , 1.  ],
+        [4.  , 2.  , 1.  ]])
+ >>> np.vander([1, 2, 3, 4], 2)
+ array([[1, 1],
+        [2, 1],
+        [3, 1],
+        [4, 1]])
+ >>> np.vander((1, 2, 3, 4), 4)
+ array([[ 1,  1,  1,  1],
+        [ 8,  4,  2,  1],
+        [27,  9,  3,  1],
+        [64, 16,  4,  1]])
+ 
+3 - general ndarray creation functions
+-----------------------------
+
+The ndarray creation functions e.g. :func:`numpy.ones`,
+:func:`numpy.zeros`, and `default_rng().random
+<random.Generator.random>` define arrays based upon the desired shape.
+The  ndarray creation functions can create arrays with any dimension by
+specifying how many dimensions and length along that dimension in a
+tuple or list. 
+
+:func:`numpy.zeros` will create an array filled with 0 values with the specified
+shape. The default dtype is ``float64``::
 
-indices() will create a set of arrays (stacked as a one-higher dimensioned
-array), one per dimension with each representing variation in that dimension.
-An example illustrates much better than a verbal description: ::
+ >>> np.zeros((2, 3))
+ array([[0., 0., 0.], 
+        [0., 0., 0.]])
+ >>> np.zeros((2, 3, 2))
+ array([[[0., 0.],
+         [0., 0.],
+         [0., 0.]],
+
+        [[0., 0.],
+         [0., 0.],
+         [0., 0.]]])
+
+:func:`numpy.ones` will create an array filled with 1 values. It is identical to
+``zeros`` in all other respects as such::
+
+ >>> np.ones((2, 3))
+ array([[ 1., 1., 1.], 
+        [ 1., 1., 1.]])
+ >>> np.ones((2, 3, 2))
+ array([[[1., 1.],
+         [1., 1.],
+         [1., 1.]],
+
+        [[1., 1.],
+         [1., 1.],
+         [1., 1.]]])
+
+:func:`default_rng.random` will create an array filled with random
```

Comment:
I think this will need to be something like:
```suggestion
The :meth:`~numpy.random.Generator.random` method of the result of :func:`default_rng` will create an array filled with random
```
for the link to work